### ISSUE TITLE
TNFW profile not differentiable w.r.t. Rs, center_x, center_y

### ISSUE THREAD

**Issue by ahuang314** [lenstronomy/JAXtronomy#66]

**Comment by ahuang314** [comment]
Addressed in PR #97.

In general, whenever `jnp.where(bool, branch_1, branch_2)` is executed, both branches are computed and one is discarded depending on the bool. However, when computing gradients, any NaNs that appear in either branch are propagated forwards, even if that branch is discarded.
